storyscript/storyscript
# feat(parser): accept n-number of arguments in dot mutations

**- What I did**

Dot mutations had been restricted to accepting exactly one argument
which is not suitable in situations where the mutation might not
need any arguments or may require more than one argument.
Before this commit, these examples and similar cases used to fail
the parsing phase since the grammar itself didn't allow for
variable number of arguments for dot mutations.

**- How I did it**

With this commit we change that and modify grammar to allow such
cases and adjust the lowering phase to handle multiple arguments
gracefully.

**- How to verify it**

These should work fine now:

1:	[1, 2].length()                   # No argument required.
2:	[1, 2, 3].replace(item: 2 by: 4)  # Two arguments required.

Closes: #959.


PS: This PR contains a commit from another PR #1014. It isn't exactly part of this PR but since my other PR is waiting a merge and this PR has some overlapping changes, I pushed that commit as well in this PR so that we can at least start a discussion about the main issue behind this PR. Commit message header for out of place commit: feat(parser): stop accepting old (whitespace+then) mutation syntax
